src/wolframite/jlink.clj

```diff
@@ -102,32 +111,37 @@
       (->> (mapv parse-long))))
 
 (defn- version-path
-  "For installation that use the format <base path>/<version>, detect full path to the latest one
+  "Detects type of path, i.e. whether or not there is a version number subdirectory, and returns the relevant one.
 
+  TODO: Functions more general than jlink should not be in this namespace. Maybe make a 'system' namespace?
   NOTE:
-  - Currently, not being used?
   - Returns a string.
   "
   [base-path]
-  (if-let [version-dir
-           (some->> (when (fs/exists? base-path)
-                      (fs/list-dir base-path))
-                    (sort-by version-vector)
-                    last)]
-    version-dir
-    (throw
-     (ex-info (str "Could not find a Wolfram version directory (directory with numerical values separated by '.') at the given base path.") {:paths base-path}))))
-
-(defn ->platform-id
-  "Coerces to a common platform identifier or throws an 'unrecognised' error."
-  [platform]
+
+  (let [path (when (fs/exists? base-path) (str base-path))
+        version-dir
+        (some->> path
+                 fs/list-dir
+                 (sort-by version-vector)
+                 last
+                 str)]
+    (if version-dir version-dir
+        (if path
+          (throw
+           (ex-info (str "Could not find a Wolfram base directory (directory with numerical values separated by '.') at the given base path.")
+                    {:paths base-path}))))))
+
+(defn- ->platform-id
+  "Coerces to a common os identifier or throws an 'unrecognised' error."
+  [os]
```

Comment:
This is partially a legacy of the old system, but  I do think keywords make for better enums. The coercion only needs to happen once at the edge and the three valid options can be decided internally and otherwise used consistently throughout.